### ISSUE TITLE
Address S6967 FP: Raises when model is validated through helper method

### ISSUE THREAD

**Issue by RobSiklos** [SonarSource/sonar-dotnet#9325]
### Description

For rule S6967, we're getting a lot of false positives, because our controller actions call a helper method which does the validation.

We'd like the ability to somehow decorate or otherwise identify this method, so that we can indicate to the analyzer that the model state is indeed validated.

### Repro steps

Simplified example: 

```csharp
[Controller]
public class MyController : ControllerBase
{
    [HttpGet]
    public IActionResult DoSomething()
    {
        return this.DoAction(() => {
            return this.Ok();
        });
    }

    [ValidatesModelState]
    private IActionResult DoAction(Func<IActionResult> action)
    {
        if (!this.ModelState.IsValid)
        {
            return this.BadRequest();
        }

        return action();
    }

    [System.AttributeUsage(AttributeTargets.Method)]
    private sealed class ValidatesModelStateAttribute : Attribute { }
}
```

### Expected behavior

Because I decorated the `DoAction` method with an attribute called `ValidatesModelStateAttribute`, the warning is not triggered if `DoAction` is called in the controller action.

### Actual behavior

S6967 is triggered for the `DoSomething()` method, and there is no way to tell the analyzer that I am using the `DoAction` method to validate the model state.

### Related information

* .NET 6.0

**Comment by mary-georgiou-sonarsource** [comment]
Hello @RobSiklos, 

I confirm this is a false positive. I'll add it to our backlog to deal with soon as we're currently working on a hardening for these rules.

Thanks a lot!

**Comment by RobSiklos** [comment]
Thanks @zsolt-kolbay-sonarsource , it looks like that solution of using the attribute may have done the trick.

**Comment by bgxie** [comment]
Using the Validation Attribute on the Controllers makes things workable. 

It still requires each Controller to be decorated though. It would be great if global registrations of the attribute could be understood:

`   services.AddControllersWithViews(options =>
   {
       options.Filters.Add(new ValidateModelAttribute());
   });`

**Comment by RobSiklos** [comment]
@bgxie If you have a custom base class for all your controllers, you can just add the attribute there, and then all your controllers will inherit it.

**Comment by bgxie** [comment]
Thanks @RobSiklos That worked.
